OpenSim/Simulation/Model/Model.cpp

```diff
@@ -656,38 +666,31 @@ void Model::extendConnectToModel(Model &model)
             Joint* free = new FreeJoint(jname, *ground, *child);
             free->upd_reverse() = mob.isReversedFromJoint();
             addJoint(free);
+            setNextSubcomponentInSystem(*free);
```

Comment:
That's a good point. It currently would serialize the FreeJoint (having been added to the JointSet), so I agree adopting directly would not have that affect (like the hidden Welds that are added to break loops). I will update.